pkg/internal/config/config.go

```diff
@@ -120,64 +120,88 @@ func New(baseConfig kconfig.Config) (Config, error) {
 }
 
 func setOptionalParams(baseConfig kconfig.Config, newConfig Config) (Config, error) {
-	if rawTrafficGeneratorPacketsPerSecondInMillions :=
-		baseConfig.Params[TrafficGeneratorPacketsPerSecondInMillionsParamName]; rawTrafficGeneratorPacketsPerSecondInMillions != "" {
-		trafficGeneratorPacketsPerSecondInMillions, err := strconv.Atoi(rawTrafficGeneratorPacketsPerSecondInMillions)
-		if err != nil || trafficGeneratorPacketsPerSecondInMillions < 0 {
-			return Config{}, ErrInvalidTrafficGeneratorPacketsPerSecondInMillions
+	var err error
+
+	if rawVal := baseConfig.Params[TrafficGeneratorPacketsPerSecondInMillionsParamName]; rawVal != "" {
+		newConfig.TrafficGeneratorPacketsPerSecondInMillions, err = parseTrafficGeneratorPacketsPerSecondInMillions(rawVal)
+		if err != nil {
+			return Config{}, err
 		}
```

Comment:
Please consider the following suggestion:
```go
newConfig.TrafficGeneratorPacketsPerSecondInMillions, err = parseTrafficGeneratorPacketsPerSecondInMillions(baseConfig.Params[TrafficGeneratorPacketsPerSecondInMillionsParamName])
if err != nil {
    return Config{}, err
}
```

It doesn't matter if `rawVal` is empty, because `parseTrafficGeneratorPacketsPerSecondInMillions()` will fail anyway if its input is an empty string.


Same for the other usages of this pattern in`setOptionalParams()`.

Reply:
but we don't want them to fail if `baseConfig.Params[TrafficGeneratorPacketsPerSecondInMillionsParamName]` is empty - these are optional params so if thery are empty then we would use the default.

Reply:
You are right.
Thank you for the explanation.